Src/Base/AMReX_CArena.cpp

```diff
@@ -60,6 +60,28 @@ CArena::alloc_protected (std::size_t nbytes)
         }
     }
 
+#ifdef AMREX_USE_GPU
+    if (free_it == m_freelist.end()) {
+        // Clear Gpu::freeAsync buffer and try again.
+        // We need to unlock the mutex for this so that free() can be called.
+        // This may invalidate free_it.
+        carena_mutex.unlock();
+        const bool freed_something = Gpu::clearFreeAsyncBuffer();
+        carena_mutex.lock();
+
+        if (freed_something) {
```

Comment:
Even if nothing is freed by `Gpu:clearFreeAsyncBuffer`, we may still want to check again, because `carena_mutex` was unlocked and during that time things might have changed.

Reply:
And `wait_list_size()` could be outdated anyway.